src/pages/api/ask-ai.ts

```diff
@@ -0,0 +1,72 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+import { connectToDatabase } from '@/services/mongodb'
+import { CACHE_DURATION, connectToCacheDatabase } from '@/services/upstash'
+import { askAI } from '@/services/openai'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  try {
+    if (req.method !== 'POST') {
+      return res.status(405).end()
+    }
+
+    if (!req.body.codeSnippet) {
+      return res.status(400).json({
+        message: 'Missing code snippet',
+      })
+    }
+
+    if (!req.body.id) {
+      return res.status(400).json({
+        message: 'Missing ID',
+      })
+    }
```

Comment:
might add zod to the `Tech Debt` section on README

Reply:
Also, it'd be nice to have a error handler in the client side. Will do it. Tks!

Reply:
Added to tech debt list :)